src/Microsoft.Data.Analysis/StringDataFrameColumn.cs

```diff
@@ -17,15 +18,19 @@ namespace Microsoft.Data.Analysis
     /// <remarks> Is NOT Arrow compatible </remarks>
     public partial class StringDataFrameColumn : DataFrameColumn, IEnumerable<string>
     {
+        //The maximum size in any single dimension for array containing other types than byte or single byte structure is 0X7FEFFFFF - 2146435071
+        //See https://learn.microsoft.com/en-us/dotnet/framework/configure-apps/file-schema/runtime/gcallowverylargeobjects-element
+        public static int MaxCapacity = 2146435071 / Unsafe.SizeOf<IntPtr>(); // Size of pointer
```

Comment:
I'm not connecting the dots here.

Reply:
(Thinking Out Loud)

We're using List<string> below.

That is backed by an array:
https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/List.cs#L25C13-L25C13

Reply:
Why don't we just use this?

https://github.com/dotnet/runtime/blob/82fee2692b3954ba8903fa4764f1f4e36a26341a/src/libraries/System.Private.CoreLib/src/System/Array.cs#L2135

Reply:
https://github.com/dotnet/runtime/issues/31366

Reply:
This property is available in .Net 6.0 and later versions of .Net. As we still support Net Standard 2.0 and .Net Framework 4.6.1 ( for example our unit tests for 4.6.1 fails on attempt to create array larger than 2 Gb) it still required to avoid allocating more than 2 Gb. 

Another solution, that I thought about was to define ArrayMaxCapacityInBytes value as a global const and use the nearest power of 2 and not this "magic" value. It's practicaly 2 times less, however will guarantee that our Buffers are multiple (it helps in some cases when we work with spans of different types, for example boolean and int in ElementWise operations - we may be sure that the bound between different validybuffer chunks will not be in the middle of other type span) and in theory indexing can be faster as compiler may optimised % operation if right operand is power of two. 

But as we are going to drop support of Net Standard 2.0 and full .net Frameworks and code will be eliminated.

Reply:
@ericstj @eerhardt @tannergooding - this feels like something you guys will just know the answer to :)

Reply:
So long as this target netstandard2.0 I don't see a problem with it having a copy of a constant to represent `Array.MaxLength`  You might want to place that as a separate constant (`ArrayUtilities.MaxLength`) and mention you're polyfilling that API, then use the same constant both here and in `VBufferDataFrameColumn`.

Reply:
Fixed